src/core/components/scroll-to-top.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from "react"
+
+const ScrollToTopButton = () => {
+  const [isVisible, setIsVisible] = useState(false)
+  const [hideTimeout, setHideTimeout] = useState(null) 
```

Comment:
consider using `useRef` insted of `useState` to store the timeout ID. This way, it will persist across renders without triggering unnecessary re-renders.